### PR TITLE
Remove all arguments from transport Start

### PR DESCRIPTION
**Pull request by kriskowal**
These changes first remove the dependencies argument from inbounds and outbounds, since they are made superfluous in #504.

Then, this removes the ServiceDetail argument from Start, and replaces it with SetRegistry (needed to satisfy the Inbound contract, so Dispatcher can call it) and WithRegistry (because it is convenient to chain when constructing an inbound inline).

### CODE REVIEW

File: transport/transporttest/clientconfig.go [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
```
Comment (willhug) [inline transport/transporttest/clientconfig.go]: you can rebase, this has already been fixed! :D

File: Makefile [hunk 2]
```diff
@@ -32,6 +32,7 @@ build:
 
 .PHONY: generate
 generate:
+	go install ./vendor/golang.org/x/tools/cmd/stringer
```
Comment (willhug) [inline Makefile]: is this just something we missed before?

File: internal/examples/thrift/keyvalue/gen.go [hunk 3]
```diff
@@ -21,4 +21,4 @@
 package keyvalue
 
 //go:generate thriftrw --plugin=yarpc kv.thrift
-//go:generate ../../../scripts/updateLicenses.sh
+//go:generate ../../../../scripts/updateLicenses.sh
```
Comment (willhug) [inline internal/examples/thrift/keyvalue/gen.go]: etw67

File: transport/http/inbound.go [hunk 4]
```diff
@@ -63,12 +64,30 @@ func (i *Inbound) WithTracer(tracer opentracing.Tracer) *Inbound {
...
+func (i *Inbound) SetRegistry(registry transport.Registry) {
+	i.registry = registry
+}
+
+// Start starts the inbound with a given service detail, opening a listening
+// socket.
+func (i *Inbound) Start() error {
+
+	if i.registry == nil {
+		panic("could not start HTTP inbound without a configured registry")
```
Comment (willhug) [inline transport/http/inbound.go]: bit dangerous, let's just hope we have no composable/dynamic inbounds

File: glide.yaml [hunk 5]
```diff
@@ -5,6 +5,8 @@ import:
   subpackages:
   - context
   - context/ctxhttp
+- package: golang.org/x/tools
```
Comment (willhug) [inline glide.yaml]: is this used?

File: Makefile (hunk 2 above)
Comment (kriskowal) [inline Makefile]: yes

File: glide.yaml (hunk 5 above)
Comment (kriskowal) [inline glide.yaml]: this is where the stringer codegen tool comes from

File: Makefile (hunk 2 above)
Comment (kriskowal) [inline Makefile]: specifically, this being missing causes make generate to fail. it works for folks who have manually installed stringer globally. this is a failure of golang to isolate projects.

**Comment by kriskowal** [comment]
@willhug panics removed. ready for another round.

**Pull request merged**